lib/superform/rails.rb

```diff
@@ -344,18 +334,33 @@ def false_option(&)
           option(selected: field.value == false, value: false.to_s, &)
         end
 
+        protected
+
+          def map_options(collection)
+            OptionMapper.new(collection)
+          end
+      end
+
+      class MultipleSelectField < SelectField
+        def template(...)
+          # The HTML specification says when multiple parameter passed to select
+          # and all options got deselected web browsers do not send any value to
+          # server. Unfortunately this introduces a gotcha: if a User model has
+          # many roles and have role_ids accessor, and in the form that edits
+          # roles of the user the user deselects all roles from role_ids
+          # multiple select box, no role_ids parameter is sent.
+          input(type: "hidden", name: dom.name, value: "")
+          super(...)
```

Comment:
I also used the argument forwarding, but i realized that this was introduced in ruby 2.7 so then it needs to bump required_ruby_version in the gemspec. If you don't want to include that in this ticket then switch to *, **, & to catch all arguments and can forward explicit, or implicitly super without parenthesis. For these discussions we could include more code style guidance (e.g. rubocop)?